packages/claims/src/index.js

```diff
@@ -0,0 +1,102 @@
+import {
+  bufferToHex,
+  ecrecover,
+  ecsign,
+  fromRpcSig,
+  pubToAddress,
+  sha3,
+  toChecksumAddress,
+  toRpcSig,
+} from 'ethereumjs-util'
+
+/**
+ * Create a valid META Claim object to verify a new claim
+ *
+ * @param  {Object} account            Ethereum account object
+ * @param  {String} account.address    Account Ethereum address
+ * @param  {String} account.privateKey Account private key
+ * @param  {Object} claimMessage       Raw claim message
+ * @param  {Object} extraData          Any extra properties to add to claim object
+ * @return {Object}                    META Claim object
+ */
+export const createMetaClaimObject = (
+  account,
+  claimMessage,
+  extraData = {}
+) => {
+  return Object.assign(
+    {
+      address: account.address,
+      claimHash: bufferToHex(sha3(claimMessage)),
+      claimMessage: claimMessage,
+      signature: signMessage(claimMessage, account.privateKey),
+    },
+    extraData
+  )
+}
+
+/**
+ * Recover address from claim hash
+ *
+ * @param  {String} claimHash sha3 hash of claim message
+ * @param  {String} signature Ethereum RPC signature object
+ * @return {String}           Recovered Ethereum public address hex
+ */
+export const recoverAddressFromClaim = (claimHash, signature) => {
+  // generate signature parameters
+  const { v, r, s } = fromRpcSig(signature)
+
+  // generate claim buffer from claim hash minus `0x` prefix
+  const claimBuffer = Buffer.from(claimHash.substring(2), 'hex')
+
+  // recover public key from claim
+  const recoveredPublicKey = ecrecover(claimBuffer, v, r, s)
+
+  // generate Ethereum address hex from public key
+  const recoveredAddress = toChecksumAddress(
+    bufferToHex(pubToAddress(recoveredPublicKey))
+  )
+
+  return recoveredAddress
+}
+
+/**
+ * Generate a valid ECDSA signature for use with Ethereum RPC clients
+ *
+ * @param  {String} message    Message to sign
+ * @param  {String} privateKey Private key to sign message with
+ * @return {String}            ECDSA signature
+ */
```

Comment:
I believe there should be two options for signatures:

1) decrypt private key, and keep it unencrypted in memory until timeout (timeout can be 0 = forever)
2) private key only gets decrypted for the specific signature task, and immediately encrypted thereafter.

The risk of private key exposure is a function of the time it's plainly exposed.

Reply:
@nolash Would this not be a concern of the application using the library rather than the library itself?